pandas/tests/io/json/test_readlines.py

```diff
@@ -252,3 +252,30 @@ def test_readjson_lines_chunks_fileurl(datapath):
     with pd.read_json(file_url, lines=True, chunksize=1) as url_reader:
         for index, chuck in enumerate(url_reader):
             tm.assert_frame_equal(chuck, df_list_expected[index])
+
+
+def test_chunksize_is_incremental():
+    jsonl = (
+        """{"a": 1, "b": 2}
+        {"a": 3, "b": 4}
+        {"a": 5, "b": 6}
+        {"a": 7, "b": 8}\n"""
+        * 1000
+    )
+
+    class MyReader:
+        def __init__(self, contents):
+            self.read_count = 0
+            self.stringio = StringIO(contents)
+
+        def read(self, *args):
+            self.read_count += 1
+            return self.stringio.read(*args)
+
+        def __iter__(self):
+            self.read_count += 1
+            return iter(self.stringio)
+
+    reader = MyReader(jsonl)
+    assert len(list(pd.read_json(reader, lines=True, chunksize=100))) > 1
```

Comment:
can you be more precise, e.g. this would have 40 pieces i think

Reply:
I don't think we want to constrain the implementation that it never does read-ahead. The important thing is that it doesn't read the full thing in one go.